### ISSUE TITLE
Does this support Linux?

### ISSUE THREAD

**Issue by Zerophase** [dotBunny/CLionSourceCodeAccess#14]
Currently, I'm using codelite and would like to move to CLion.  I just want to check if this is known to work on 4.11.2 in Linux.

Also is this pull request required to get the Accessor working?
https://github.com/EpicGames/UnrealEngine/pull/2516

**Comment by reapazor** [comment]
This is mean to work without any modifications to the engine, as long as you can set the engine to output to a CodeLite normal output ... it parses that output into CMake format.

I'm not sure entirely, I havent tested it .. .but it should? Let me know and ill fix accordingly.

**Comment by Zerophase** [comment]
Alright, I'll get back to you on this within a day or two. For the CodeLite output do you mean I need CodeLite installed?

**Comment by reapazor** [comment]
I dont think so, its using Unreal internal generator so you should be good without it installed.

**Comment by Zerophase** [comment]
Yeah, CodeLite was causing issues with needing Clang 3.8 installed, while 4.11.2 needs 3.5 installed. I have to wait on the engine to recompile.

**Comment by reapazor** [comment]
Heh i havent actually tried the plugin with a source build ... this could be interesting, but i will happily make it work :)

**Comment by Zerophase** [comment]
The last guy that was working on an accessor had something up, but has since deleted the repo.

**Comment by Zerophase** [comment]
Alright, the plugin finds all of the executables needed without issue, except for Clion's. I had to manually set the path to Clion with /opt/clion/bin/clion.sh.

When I open a source class by clicking edit clion opens a window for the header, and another window for the cpp file. It doesn't give me the whole project. Going to see if closing unreal and opening with Clion support turned on works any differently.

I can compile my changes from Unreal, but can't seem to compile in CLion.

**Comment by reapazor** [comment]
Did you generate the project files ...
FIle -> Generate CMakeList?

Compiling is still not 'commited' as working yet, CLion is missing a few features that im working around. I have a hack version locally, Ill see about getting compiling outside the editor in soon.

**Comment by Zerophase** [comment]
Hmm, I just have "Refresh an IDE to edit source code project" under File. (This is all being run from within the project) Linux doesn't have the Unreal version Launcher yet) I did run "Refresh an IDE...".  No File -> Generate CMakeList shows up.

CLion ends up opening each file as a separate project in the folder /tmp/FileName.ext, and all of the project information points to where that file exists. Not sure if opening the project in /tmp is supposed to happen with the Accessor.

**Comment by reapazor** [comment]
Do you see any CMakeLists.txt in your project folder ?  Is there any other extra files added?
[im gonna grab a source build tmr and see what i can find]

It doesnt sound like its using the same pathways for binding as the built version does, i will look into that tonight.

**Comment by Zerophase** [comment]
Yeah, there's a CMakeLists.txt in my project folder. 
I also have a: 
- Makefile
- .kdev4 
- .pro 
- .uproject 
- .workspace 
- CodeCompletionFolders.txt 
- CodeLitePreprocessor.txt 
- Config.pri 
- Defines.pri 
- Header.pri 
- Includes.pri
- Source.pri

**Comment by Zerophase** [comment]
#13

**Comment by Zerophase** [comment]
#13 seems to partially effect the Linux project. I tested without spaces, and was able to get the header and cpp file to open in the same CLion window sometimes. CLion still can't find the CMakeLists.txt, though.

**Comment by reapazor** [comment]
There generally seems to be a problem with spaces in folder names (that extends to unreal as well). I've gotta do a pass on another issue to deal with spaces in path names and how to handle them, this might get lumped in with that when I find a solution.

**Comment by reapazor** [comment]
I think i may have fixed it all with the fixes to spaces etc in paths pushes of late :)
